src/pages/Homepage.jsx

```diff
@@ -1,70 +1,83 @@
-import { useState } from 'react';
+import { useCallback, useState } from 'react';
+import { v4 as uuid } from 'uuid';
 import TimerComponent from '../components/TimerComponent';
 import ActiveTeamsComponent from '../components/ActiveTeamsComponent';
 import EliminatedTeamsComponent from '../components/EliminatedTeamComponent';
+import findTeam from '../utils/index.js';
 
 const Homepage = () => {
   const [teams, setTeams] = useState([]);
   const [draggedIndex, setDraggedIndex] = useState(null);
 
   const handleAddTeam = (type) => {
-    setTeams([
-      ...teams,
+    setTeams((prevState) => [
+      ...prevState,
       {
+        id: uuid(),
         name: '',
         score: 0,
-        active: type === 'active' ? true : false,
+        active: type === 'active',
       },
     ]);
   };
 
-  const handleNameInput = (e, index) => {
+  const handleNameInput = useCallback((event) => {
```

Comment:
Always use the `useCallback` when dealing with event handlers. This lowers the memory footprint and makes the component more robust.

Reply:
Alright, another use case, noted.